src/components/WalletKit/ConnectWallet.tsx

```diff
@@ -12,59 +12,108 @@ import { ConnectedModal } from "@/components/WalletKit/ConnectedModal";
 import { WalletKitContext } from "@/components/WalletKit/WalletKitContextProvider";
 
 import { trackEvent, TrackingEvent } from "@/metrics/tracking";
+import { localStorageSavedWallet } from "@/helpers/localStorageSavedWallet";
 
 export const ConnectWallet = () => {
   const { network, walletKit, updateWalletKit } = useStore();
+  const [connected, setConnected] = useState<boolean>(false);
   const [isModalVisible, setShowModal] = useState(false);
   const [errorMessageOnConnect, setErrorMessageOnConnect] = useState("");
   const walletKitInstance = useContext(WalletKitContext);
+  const savedWallet = localStorageSavedWallet.get();
 
   const { data: accountInfo, refetch: fetchAccountInfo } = useAccountInfo({
     publicKey: walletKit?.publicKey || "",
     horizonUrl: network?.horizonUrl || "",
     headers: network ? getNetworkHeaders(network, "horizon") : {},
   });
 
-  const resetWalletKit = () => {
+  const disconnect = () => {
     updateWalletKit({
       publicKey: undefined,
       walletType: undefined,
     });
 
     setShowModal(false);
+    setConnected(false);
+    localStorageSavedWallet.remove();
   };
 
+  useEffect(() => {
+    if (
+      !connected &&
+      savedWallet?.id !== undefined &&
+      savedWallet?.id !== "false" &&
+      savedWallet?.id !== "wallet_connect"
```

Comment:
Could we write it like this? `savedWallet?.id` can only have one value, so it might be clearer.

```suggestion
      ![undefined, "false", "wallet_connect"].includes(savedWallet?.id)
```